### DISCUSSION TITLE
Need help with enabling text-to-speech, "Failed to generate speech: Voicemessagesforbidden"

### DISCUSSION THREAD

**Discussion by SQLimit** [n3d1117/chatgpt-telegram-bot#503] [Q&A] [answered]
Hi, I send the bot with "/tts my house" command, but it said "Failed to generate speech: Voicemessagesforbidden"
Am I wrong with .env file or something else, other request like /image is working normally.

```
# Optional configuration, refer to the README for more details
# BUDGET_PERIOD=monthly
# USER_BUDGETS=*
# GUEST_BUDGET=100.0
# TOKEN_PRICE=0.002
# IMAGE_PRICES=0.016,0.018,0.02
# TRANSCRIPTION_PRICE=0.006
# VISION_TOKEN_PRICE=0.01
 ENABLE_QUOTING=true
 ENABLE_IMAGE_GENERATION=true
 ENABLE_TTS_GENERATION=true
 ENABLE_TRANSCRIPTION=true
 ENABLE_VISION=true
# PROXY=http://localhost:8080
 OPENAI_MODEL=gpt-4-vision-preview
# OPENAI_BASE_URL=https://example.com/v1/
 ASSISTANT_PROMPT="You are a helpful assistant."
# SHOW_USAGE=false
# STREAM=true
# MAX_TOKENS=1200
# VISION_MAX_TOKENS=300
# MAX_HISTORY_SIZE=15
# MAX_CONVERSATION_AGE_MINUTES=180
 VOICE_REPLY_WITH_TRANSCRIPT_ONLY=true
 VOICE_REPLY_PROMPTS="Hi bot;Hey bot;Hi chat;Hey chat"
 VISION_PROMPT="What is in this image"
# N_CHOICES=1
# TEMPERATURE=1.0
# PRESENCE_PENALTY=0.0
# FREQUENCY_PENALTY=0.0
 IMAGE_MODEL=dall-e-3
 IMAGE_QUALITY=hd
 IMAGE_STYLE=natural
 IMAGE_SIZE=1024x1024
 IMAGE_FORMAT=document
 VISION_DETAIL=low
# GROUP_TRIGGER_KEYWORD=""
# IGNORE_GROUP_TRANSCRIPTIONS=true
# IGNORE_GROUP_VISION=true
 TTS_MODEL=tts-1
 TTS_VOICE=alloy
# TTS_PRICES=0.015,0.030
# BOT_LANGUAGE=en
 ENABLE_VISION_FOLLOW_UP_QUESTIONS=true
 VISION_MODEL=gpt-4-vision-preview
```

**Comment by alexk-epsilon** [comment]
Hi @SQLimit. I tried duplicating your issue using the .env file above but /tts worked just fine for me. On the other hand, I don't have access to gpt-4-vision-preview (not sure if it matters). Could it be you OpenAI account settings or perhaps the code is different from baseline on your end ?
![Screenshot_20240118-123033](https://github.com/n3d1117/chatgpt-telegram-bot/assets/150628561/099c4628-a057-49b0-9ed0-d6a9b9036baf)

**Reply by SQLimit** [reply]
I prefer the issue might relate with the OpenAI settings, but I couldn't find out, I would figure it out when I had time.

**Comment by momai** [comment]
I'm experiencing the same problem on different openAI models.

**Comment by momai** [comment, marked as answer]
Oh, turns out I'm a fool. I had audio messages disabled in Telegram. That's why I was getting that error.
@SQLimit, check mate, maybe you just have an audio message ban in Telegram too.

**Reply by SQLimit** [reply]
Yes, after enable the audio message, it works well!
